File: src/test/java/no/utgdev/diy/ex1/Ex1Test.java
package no.utgdev.diy.ex1;

import no.utgdev.diy.DIYStatic;
import org.junit.jupiter.api.Test;

import java.lang.reflect.Method;
import java.util.List;
import java.util.Set;

import static java.util.stream.Collectors.toList;
import static org.assertj.core.api.Assertions.assertThat;

class Ex1Test {
    @Test
    void finnAlleBeans() {
        List<String> methods = methodNames(DIYStatic.scan("no.utgdev.diy.ex1"));

        assertThat(methods).hasSize(4);
        assertThat(methods).containsExactlyInAnyOrder("number", "string", "list", "map");
    }

    @Test
    void finnAlleBeansFraRoot() {
        List<String> methods = methodNames(DIYStatic.scan(Ex1Config.class));

        assertThat(methods).hasSize(4);
        assertThat(methods).containsExactlyInAnyOrder("number", "string", "list", "map");
    }

    private static List<String> methodNames(Set<Method> methods) {
        return methods
                .stream()
                .map(Method::getName)
                .collect(toList());
    }
}

File: src/test/kotlin/no/utgdev/diy/ex1/Ex1Test.kt
package no.utgdev.diy.ex1

import no.utgdev.diy.BeanDefinition
import no.utgdev.diy.DIYStatic.scan
import org.junit.jupiter.api.Test
import org.assertj.core.api.Assertions
import java.lang.reflect.Method
import java.util.stream.Collectors

internal class Ex1Test {
    @Test
    fun finnAlleBeans() {
        val methods = methodNames(scan("no.utgdev.diy.ex1"))
        Assertions.assertThat(methods).hasSize(4)
        Assertions.assertThat(methods).containsExactlyInAnyOrder("number", "string", "list", "map")
    }

    @Test
    fun finnAlleBeansFraRoot() {
        val methods = methodNames(scan(Ex1Config::class.java))
        Assertions.assertThat(methods).hasSize(4)
        Assertions.assertThat(methods).containsExactlyInAnyOrder("number", "string", "list", "map")
    }

    companion object {
        private fun methodNames(methods: Set<BeanDefinition>): List<String> {
            return methods
                .stream()
                .map { obj -> obj.name }
                .collect(Collectors.toList())
        }
    }
}